corehq/apps/reports/standard/cases/case_data.py

```diff
@@ -555,6 +559,209 @@ def close_case_view(request, domain, case_id):
+                case_names[case.case_id] += ' <span class="label label-default">primary case</span>'
+        case_xforms = case.xform_ids
+
+        for form_id in case_xforms:
+            if form_id not in delete_forms:
+                delete_forms.insert(0, form_id)
+            form_object = XFormInstance.objects.get_form(form_id, domain)
+            if form_id not in form_names:
+                form_names[form_id] = xmlns_to_name(domain, form_object.xmlns, form_object.app_id)
+                if form_names[form_id] == form_object.xmlns:
+                    form_name = [
+                        get_app(domain, form_object.app_id).name or "[Unknown App]",
+                        "[Unknown Module]",
+                        form_object.name or "[Unknown Form]"
+                    ]
+                    form_names[form_id] = ' > '.join(form_name)
+            case_db = FormProcessorInterface(domain).casedb_cache(
+                domain=domain,
+                load_src="process_stock",
+            )
+            touched_cases = FormProcessorInterface(domain).get_cases_from_forms(case_db, [form_object])
+
+            case_actions = {}
+            for touched_id in touched_cases:
+                case_object = safely_get_case(request, domain, touched_id)
+                actions = list(touched_cases[touched_id].actions)
+                if touched_id == case.case_id:
+                    case_actions['current'] = actions
+                elif touched_id not in delete_cases:
+                    if touched_id not in case_actions:
+                        case_actions[case_object.name] = actions
+                    if const.CASE_ACTION_CREATE in actions and touched_id != case.case_id:
+                        subcase_count += 1
+                        get_forms_for_deletion_from_case(case_object, subcase_count)
+                        subcase_count -= 1
+                    if const.CASE_ACTION_CLOSE in actions:
+                        reopened_cases[touched_id] = form_id
+                        case_names[touched_id] = case_object.name
+                    if any(action in actions for action in update_actions):
+                        if touched_id not in affected_cases:
+                            affected_cases[touched_id] = {}
+                        affected_cases[touched_id][form_id] = ', '.join(actions)
+                        case_names[touched_id] = case_object.name
+            cases[case.case_id][form_id] = case_actions
+        return
+
+    case_instance = safely_get_case(request, domain, case_id)
+    subcase_count = 0
+    try:
+        get_forms_for_deletion_from_case(case_instance, subcase_count)
+    except ValueError:
+        messages.error(request, _("Deleting this case would delete too many related cases. "
+                                  "Please delete some of this cases' subcases before attempting"
+                                  "to delete this case."))
+        return {}, True
```

Comment:
What do you think of including the "redirect" return code in the dict?
```suggestion
        return {"redirect": True}
```